test/integration/apiserver/certreload/certreload_test.go

```diff
@@ -253,3 +339,89 @@ YkNtGc1RUDHwecCTFpJtPb7Yu/E=
 		}
 	}
 }
+
+func TestSNICert(t *testing.T) {
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+
+	var servingCertPath string
+
+	_, kubeconfig := framework.StartTestServer(t, stopCh, framework.TestServerSetup{
+		ModifyServerRunOptions: func(opts *options.ServerRunOptions) {
+			opts.GenericServerRunOptions.MaxRequestBodyBytes = 1024 * 1024
+			servingCertPath = opts.SecureServing.ServerCert.CertDirectory
+
+			if err := ioutil.WriteFile(path.Join(servingCertPath, "foo.key"), anotherServerKey, 0644); err != nil {
+				t.Fatal(err)
+			}
+			if err := ioutil.WriteFile(path.Join(servingCertPath, "foo.crt"), anotherServerCert, 0644); err != nil {
+				t.Fatal(err)
+			}
+
+			dynamiccertificates.FileRefreshDuration = 1 * time.Second
+			opts.SecureServing.SNICertKeys = []flag.NamedCertKey{{
+				Names:    []string{"foo"},
+				CertFile: path.Join(servingCertPath, "foo.crt"),
+				KeyFile:  path.Join(servingCertPath, "foo.key"),
+			}}
+		},
+	})
+	apiserverURL, err := url.Parse(kubeconfig.Host)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// when we run this the second time, we know which one we are expecting
+	acceptableCerts := [][]byte{}
+	tlsConfig := &tls.Config{
+		InsecureSkipVerify: true,
+		VerifyPeerCertificate: func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
+			acceptableCerts = make([][]byte, 0, len(rawCerts))
```

Comment:
We make 2 connections with this tlsConfig, so doesn't the 2nd time we run it it overwrites any info found via the first connection?

What is the point of `conn` then since we don't test anything about the `acceptableCerts` until after the `conn2` is made?

Seems like after the first connection is made that you want to check that there are 0 certs?

Reply:
I'll add a test on the initial connection's cert